src/App.tsx

```diff
@@ -15,48 +23,113 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+type Goods = string[];
+
+const getSortedGoods = (
+  goods: Goods,
+  sortMethod: string,
+  isReversed: boolean,
+) => {
+  let newGoods = [...goods];
+
+  switch (sortMethod) {
+    case Sort.ALPHABETICALLY:
+      newGoods = newGoods.sort((good1, good2) => good1.localeCompare(good2));
+      break;
+    case Sort.BY_LENGTH:
+      newGoods.sort().sort((good1, good2) => good1.length - good2.length);
+      break;
+    case Sort.RESET:
+      newGoods = [...goodsFromServer];
+      break;
+    default:
+      return isReversed ? newGoods.reverse() : newGoods;
+  }
+
+  return isReversed ? newGoods.reverse() : newGoods;
+};
+
 export const App: React.FC = () => {
+  const [selectedSort, setSelectedSort] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const sortedGoods = getSortedGoods(goodsFromServer, selectedSort, isReversed);
+
+  const isResetHidden = selectedSort === '' && !isReversed;
+
+  const handleSortChange = (eventName: Sort) => {
+    if (eventName === selectedSort) {
+      return;
+    }
+
+    if (eventName === Sort.RESET) {
+      setSelectedSort('');
+      setIsReversed(false);
+
+      return;
+    }
+
+    setSelectedSort(eventName);
+  };
+
+  const handleReverse = () => {
+    setIsReversed(!isReversed);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn({
+            'button is-info': selectedSort === Sort.ALPHABETICALLY,
+            'button is-info is-light': selectedSort !== Sort.ALPHABETICALLY,
```

Comment:
```suggestion
            'is-light': selectedSort !== Sort.ALPHABETICALLY,
```

check all places please